File: testdata/py-project/src/service.py
from src.models import User, Product, create_user
from src.utils import generate_id, format_name
from typing import Optional


class UserService:
    def __init__(self):
        self.users = {}

    def create(self, name: str, email: str) -> User:
        user_id = generate_id()
        user = create_user(name, email)
        user.id = user_id
        self.users[user_id] = user
        return user

    def get(self, user_id: str) -> Optional[User]:
        return self.users.get(user_id)

File: testdata/rust-project/src/service.rs
use crate::models::{User, Status};

pub struct UserService {
    users: std::collections::HashMap<String, User>,
}

impl UserService {
    pub fn new() -> Self {
        UserService {
            users: std::collections::HashMap::new(),
        }
    }

    pub fn create_user(&mut self, name: String, email: String) -> User {
        let id = generate_id();
        let user = User { id: id.clone(), name, email };
        self.users.insert(id, user.clone());
        user
    }

    pub fn get_user(&self, id: &str) -> Option<&User> {
        self.users.get(id)
    }
}

fn generate_id() -> String {
    format!("{}", std::time::SystemTime::now()
        .duration_since(std::time::UNIX_EPOCH)
        .unwrap()
        .subsec_nanos())
}
